client/team.go

```diff
@@ -0,0 +1,64 @@
+package client
+
+import "errors"
+
+func (self *ApiClient) TeamCreate(payload TeamCreatePayload) (Team, error) {
+	if payload.Name == "" {
+		return Team{}, errors.New("Must specify team name on creation")
+	}
+	// AREL TODO: Check this
+	if payload.OrganizationId != "" {
+		return Team{}, errors.New("Must not specify organizationId")
+	}
+	organizationId, err := self.organizationId()
+	if err != nil {
+		return Team{}, err
+	}
+	payload.OrganizationId = organizationId
+
+	var result Team
+	err = self.http.Post("/teams", payload, &result)
+	if err != nil {
+		return Team{}, err
+	}
+	return result, nil
+}
+
+func (self *ApiClient) Team(id string) (Team, error) {
+	var result Team
+	err := self.http.Get("/teams/"+id, nil, &result)
```

Comment:
a little out of scope, but why there is a diff between get-team to get-teams in our API? is there a special reason that you know? 
In general I expected to see /team?id={id}&orgId{org_id}... where and empty id triggers get-all

Reply:
It's a design choice we've made. I think it's technically more RESTful but I'm not sure.
One difference, for example, is that the by-id API returns a single team (not an array), and returns 404 if there's no such team